affinity/cpp-20/d0796r2.md

```diff
@@ -170,150 +187,134 @@ This feature could be easily scaled to heterogeneous and distributed systems, as
+An **execution agent** executes work, typically implemented by a *callable*, 
+on an **execution resource** of a given **execution architecture**. 
+An **execution context** manages a set of execution agents on an 
+execution resource.
+An **executor** submits work to an execution context.
+More that one executor may submit work to an execution context.
+More than on execution context may manage execution agents
+on an execution resource.
 
-> [*Note:* Note that an execution resource is not limited to resources which execute work, but also a general resource where no execution can take place but memory can be allocated such as off-chip memory. *--end note*]
+> [*Note:* The execution context terminology used here
+and in the Networking TS [[33]][networking-ts] deviate from the 
+traditional *context of execution* usage that refers
+to the state of a single executing callable; *e.g.*,
+program counter, registers, stack frame. *--end note*]
 
-> [*Note:* The intention is that the actual implementation details of a resource topology are described in an execution context when required. This allows the execution resource objects to be lightweight objects that serve as identifiers that are only referenced. *--end note*]
+The **concurrency** of an execution resource is an upper bound of the
+number of execution agents that could concurrently make forward progress 
+on that execution resource. 
+It is guaranteed that no more than **concurrency** execution agents 
+could make concurrent forward progress; it is not guaranteed that
+**concurrency** execution agents will ever concurrently make forward progress.
+
+The **affinity** between an execution resource and memory resource
+is an upper bound of the uncontended bandwidth or latency 
+between an execution agent running on the execution resource and
+memory allocated from the memory resource.
 
-### System topology
 
-The system topology is made up of a number of system level `execution_resource`s, which can be queried through `this_system::resource` which returns a `std::vector`. The `execution_resources` available within the system can be initialised dynamically by a runtime library, however must be done so before `main` is called, given that after that point the system topology cannot change.
+### System topology
 
-Below *(Listing 2)* is an example of iterating over the system level resources and priniting out it's capabilities.
+A *system* includes execution resources and memory resources.
+Execution resources are organized hierarchically.
+A particular execution resource may be a partitioned into
+a collection of execution resources referred to as *members of*
+that execution resource.
+The partitioning of execution resources implies a locality relationship;
```

Comment:
Is it always possible to make this guarantee? Could you, for example, have an architecture which is represented with a hierarchy of {{A, B, C, D},{E, F, G, H}} but where D is more local to E than to A due to the layout of processing units on the die?

Reply:
I haven't encountered such a thing.  If I did see such a situation I'd say the topology definition was in error.